### PR TITLE
Fix issue with LinePerThreadBufferingOutputStream being reused across threads

### PR DESCRIPTION
**Pull request by ghale**
See https://github.com/gradle/gradle/issues/2635.

### CODE REVIEW

**Comment by big-guy** [comment]
This LGTM as a targeted fix. 

@oehme is this something we should also consider for 4.7 RC2? We think it's actually an old problem (#2635), but the GE folk were complaining about it more with 4.7.

File: subprojects/base-services/src/main/java/org/gradle/internal/io/LinePerThreadBufferingOutputStream.java [hunk 1]
```diff
@@ -67,9 +67,10 @@ public boolean checkError() {
 
     @Override
     public void close() {
-        PrintStream stream = maybeGetStream();
-        if (stream != null) {
-            stream.close();
+        PrintStream currentStream = maybeGetStream();
+        if (currentStream != null) {
+            currentStream.close();
+            stream.set(null);
```
Comment (oehme) [inline subprojects/base-services/src/main/java/org/gradle/internal/io/LinePerThreadBufferingOutputStream.java]: Shouldn't this be before the close call?

**Comment by oehme** [comment]
I'm fine with merging this for RC2 if it doesn't delay the RC further.

File: subprojects/base-services/src/main/java/org/gradle/internal/io/LinePerThreadBufferingOutputStream.java (hunk 1 above)
Comment (ghale) [reply subprojects/base-services/src/main/java/org/gradle/internal/io/LinePerThreadBufferingOutputStream.java]: Good point.  Done.

**Pull request merged**